tests/framework/pvc_utils.go

```diff
@@ -0,0 +1,192 @@
+	return &k8sv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        pvcName,
+			Annotations: annotations,
+			Labels:      labels,
+		},
+		Spec: k8sv1.PersistentVolumeClaimSpec{
+			AccessModes: []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
+			Resources: k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceName(k8sv1.ResourceStorage): resource.MustParse(size),
+				},
+			},
+		},
+	}
+}
+
+// Wait for the PVC to be in a particular phase (Pending, Bound, or Lost)
+func (f *Framework) WaitForPersistentVolumeClaimPhase(phase k8sv1.PersistentVolumeClaimPhase, pvcName string) error {
+	wait.PollImmediate(poll, DefaultTimeout, func() (bool, error) {
+		pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.GetName()).Get(pvcName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		} else {
+			if pvc.Status.Phase == phase {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	return fmt.Errorf("PersistentVolumeClaim %s not in phase %s within %v", pvcName, phase, DefaultTimeout)
+}
+
+// Create a Pod with the passed in PVC mounted under /pvc. You can then use the executor utilities to
+// run commands against the PVC through this Pod.
+func (f *Framework) CreateExecutorPodWithPVC(podName string, pvc *k8sv1.PersistentVolumeClaim) *k8sv1.Pod {
+	var pod *k8sv1.Pod
+	podDef := newExecutorPodWithPVC(podName, "ls -1 /pvc | wc -l", pvc)
+	err := wait.PollImmediate(2*time.Second, DefaultTimeout, func() (bool, error) {
+		var err error
+		pod, err = f.K8sClient.CoreV1().Pods(f.Namespace.GetName()).Create(podDef)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		Fail("Unable to create Pod: " + podName + ", error: " + err.Error())
+	}
+	return pod
+}
+
+// Delete the passed in Pod from the passed in Namespace
+func (f *Framework) DeletePodFromNamespace(pod *k8sv1.Pod, namespace *k8sv1.Namespace) {
+	err := wait.PollImmediate(2*time.Second, DefaultTimeout, func() (bool, error) {
+		err := f.K8sClient.CoreV1().Pods(namespace.GetName()).Delete(pod.GetName(), &metav1.DeleteOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		Fail("Unable to delete Pod: " + pod.GetName() + ", error: " + err.Error())
+	}
+}
+
+// Delete the passed in Pod from the Framework primary namespace
+func (f *Framework) DeletePod(pod *k8sv1.Pod) {
+	f.DeletePodFromNamespace(pod, f.Namespace)
+}
+
+func newExecutorPodWithPVC(podName, cmd string, pvc *k8sv1.PersistentVolumeClaim) *k8sv1.Pod {
+	return &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    "runner",
+					Image:   "fedora:28",
+					Command: []string{"/bin/sh", "-c", "sleep 5; echo I am an executor pod;"},
+					VolumeMounts: []k8sv1.VolumeMount{
+						{
+							Name:      pvc.GetName(),
```

Comment:
unless GetName solves some problem I am unaware of I don't see what's wrong w/ `pvc.Name`

Reply:
See my answers above, its my Java habit, but I don't have a strong opinion either way.